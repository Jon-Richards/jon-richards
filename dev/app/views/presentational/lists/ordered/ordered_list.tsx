import * as React from 'react';
import { List } from '../interfaces/list';
import { ListItem } from '../interfaces/list_item';

export function OrderedList(props: OrderedListProps): JSX.Element {
  const items = mapItems(props.items);
  return (
    <ol title={props.title}>
      {items}
    </ol>
  )
}

export type OrderedListProps = List<ListItem>;

function mapItems(items: ListItem[]): JSX.Element {
  const result = items.map(item => <li key={item.id}>{item.content}</li>);
  return <React.Fragment>{result}</React.Fragment>
}
