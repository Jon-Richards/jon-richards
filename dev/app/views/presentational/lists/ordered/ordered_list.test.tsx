import * as React from 'react';
import { OrderedList } from './ordered_list';
import { render } from 'enzyme';

describe('The OrderedList component.', () => {
  it('Should match the snapshot.', () => {
    const wrapper = render(
      <OrderedList
        title="An Ordered List."
        items={[
          { id: 'one', content: 'Item one' },
          { id: 'two', content: 'Item one' }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
