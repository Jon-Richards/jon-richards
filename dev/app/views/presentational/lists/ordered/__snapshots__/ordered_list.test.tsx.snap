// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The OrderedList component. Should match the snapshot. 1`] = `
<ol
  title="An Ordered List."
>
  <li>
    Item one
  </li>
  <li>
    Item one
  </li>
</ol>
`;
